lib/api/objectPut.js

```diff
@@ -40,86 +41,107 @@ function _storeIt(bucketName, objectKey, objMD, authInfo, canonicalID,
     if (request.query && request.query['x-amz-acl']) {
         headers['x-amz-acl'] = request.query['x-amz-acl'];
     }
-    const metadataStoreParams = {
-        objectKey,
-        authInfo,
-        metaHeaders,
-        size,
-        contentType: request.headers['content-type'],
-        cacheControl: request.headers['cache-control'],
-        contentDisposition: request.headers['content-disposition'],
-        contentEncoding:
-            removeAWSChunked(request.headers['content-encoding']),
-        expires: request.headers.expires,
-        headers,
+
+    registry.run({
```

Comment:
That's great. I don't like that it's in `objectPut.js` at all, but if I hadn't started there I would probably never have gotten this PR submitted :)